File: PizzaStore/src/main/java/com/luckyzhangx/pizzastore/MainActivity.java
package com.luckyzhangx.pizzastore;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import java.io.IOException;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        PizzaStore pizzaStore = new PizzaStore();
        Meal meal = pizzaStore.order("CalzonePizza");
        System.out.println("Bill: $" + meal.getPrice());
    }
}

File: Demo/src/main/java/com/example/demo/MainActivity.kt
package com.example.demo

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.recyclerview.widget.DefaultItemAnimator
import com.luckyzhangx.coreandroidlibs.renderer.renderadapter.DiffViewRendererAdapter
import com.luckyzhangx.coreandroidlibs.utils.renderer.TextViewRenderer
import com.luckyzhangx.corejavalibs.core.configure
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    val mAdapter = DiffViewRendererAdapter()

    val mList = mutableListOf<TextViewRenderer>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        configure(list) {
            adapter = this@MainActivity.mAdapter
            itemAnimator = DefaultItemAnimator().apply {
                supportsChangeAnimations = false
                val duration = 1000L
                addDuration = duration
                changeDuration = duration
                removeDuration = duration
                moveDuration = duration
            }
        }
        mAdapter.postList(mList)

        insert.setOnClickListener { mList.add(TextViewRenderer("new")) }
        remove.setOnClickListener { mList.removeAt(mList.lastIndex) }
        change.setOnClickListener { mAdapter.notifyItemChanged(0) }
        post.setOnClickListener { mAdapter.postList(mList) }

    }
}
